lib/gemstash/gem_source/private_source.rb

```diff
@@ -67,7 +67,15 @@ def serve_info(name)
       end
 
       def serve_marshal(id)
-        halt 403, "Not yet supported"
+        gem_full_name = id.sub(/\.gemspec\.rz\z/, "")
+        gem = storage.resource(gem_full_name)
+        halt 404 unless gem.exist?
+        gem.load
+        halt 403, "That gem has been yanked" unless gem.properties[:indexed]
+        spec = spec_storage.resource(gem_full_name)
+        halt 404 unless spec.exist?
+        content_type "application/octet-stream"
+        spec.load.content
       end
```

Comment:
I'll circle back on this block after I implement prereleaase specs (since that seems to be needed as I look at Bundler code and think on it more), there is commonalities here and with the specs endpoint, so factoring the commonalities among all 3 will probably simplify this a fair amount.

Regarding the headers... that's only true for the upstream gems, where we get the headers from the upstream source. The private gem headers aren't stored in storage, and I kind of think it might be better to handle that here in the `Gemstash::GemSource::PrviateSource`.

My thinking is that the PrivateSource is all about responding to web requests for private gems, so I think it makes sense for it to know about what web headers to send. If we store the headers when the private gem is pushed... it feels a bit awkward to me to construct the headers inside `Gemstash::GemPusher`. What if we want to update which headers get sent down the road? Do we only send the updated headers for new gems at that point? Do we go back and update any private gems already pushed with the new headers? Do we check if they have the header and add it dynamically before setting the headers? All this feels more complex to me than setting the headers in the PrivateSource.

Thoughts?